### PR TITLE
feat(TextareaBox): Added resizing textarea box

### PR DESCRIPTION
**Pull request by haronius**
Added resizing textarea box reusing styles and code from Textinput

Empty, Unfocused
image

Focused
image

Unfocused with text: 
image

Error
image

### CODE REVIEW

File: src/components/TextareaBox/TextareaBox.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    const value = this.state.value.slice(0, this.props.max);
+    return (
+      <TextareaBoxWrapper
+        className={className}
+        disabled={this.props.disabled}
+        >
+        <TextBoxFilled
+          onMouseUp={this.removeCancelBlur}
+          onMouseDown={this.cancelBlur}
+          onMouseLeave={this.removeCancelBlur}
```
Comment (seanevd) [inline src/components/TextareaBox/TextareaBox.js]: Are these events defined?

File: src/components/TextareaBox/TextareaBox.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          disabled={disabled}
+          rows={this.state.rows}
+          collapsed={collapsed}
+          focused={this.state.focused}>
+          <InputBoxMeasuring
+            ref={(el) => { this.measuringDivEl = ReactDOM.findDOMNode(el); }}
+            value={value}
+          >{value}</InputBoxMeasuring>
+          <InputBoxItem
+            type='text'
```
Comment (seanevd) [inline src/components/TextareaBox/TextareaBox.js]: Isn't this component supposed to be a textarea element?

File: src/components/TextareaBox/TextareaBox.js (hunk 1 above)
Comment (haronius) [reply src/components/TextareaBox/TextareaBox.js]: Yes, they're inherited

File: src/components/TextareaBox/TextareaBox.js (hunk 2 above)
Comment (haronius) [reply src/components/TextareaBox/TextareaBox.js]: Yes, I can rename it

Comment (haronius) [reply src/components/TextareaBox/TextareaBox.js]: done

Comment (seanevd) [reply src/components/TextareaBox/TextareaBox.js]: Ah yeah, it is a textarea 
image

The `type='text'` threw me off.

**Pull request merged**